File: Cpp/src/UpTime.cpp
#include <iomanip>
#include <fstream>
#include <ctime>
#include "../include/UpTime.hpp"

UpTime::UpTime(const std::string& fileName /*= "../data/uptime_file.txt"*/) {
    this->fileName = fileName;
    std::string* values_from_file = this->readFromFile();
    this->upTime = TimeFromSeconds(stoi(values_from_file[0]));
    this->idleTime = TimeFromSeconds(stoi(values_from_file[1]));
}

void UpTime::execute(std::ostream& output /*= std::cout*/) {
    std::time_t time = std::time(0);
    output << std::put_time(std::localtime(&time), "%H:%M:%S") << " up " << this->upTime.toCout() << std::endl;
    output << std::put_time(std::localtime(&time), "%H:%M:%S") << " idle " << this->idleTime.toCout();
}

std::string* UpTime::readFromFile() {
    std::string fileContent;
    std::ifstream readFile(this->fileName);
    getline(readFile, fileContent);
    readFile.close();
    static std::string toReturn[2] = {fileContent.substr(0, fileContent.find(" ")), 
                                        fileContent.substr(fileContent.find(" "))};
    return toReturn;
}

File: Ruby/lib/up_time.rb
require_relative 'time_from_seconds'

# Class to execute uptime function
class UpTime
  def initialize(file_name = '../data/uptime_file.txt')
    @file_name = File.expand_path(file_name, __dir__)
    values_from_file = read_from_file
    @up_time = TimeFromSeconds.new(values_from_file[0].to_i)
    @idle_time = TimeFromSeconds.new(values_from_file[1].to_i)
  end

  def execute(output = $stdout)
    output.puts "#{Time.now.strftime('%H:%M:%S')} up #{@up_time.to_puts}"
    output.puts "#{Time.now.strftime('%H:%M:%S')} idle #{@idle_time.to_puts}"
  end

  private

  def read_from_file
    File.read(@file_name).split
  end
end
